extensions/amp-form/0.1/form-dirtiness.js

```diff
@@ -94,13 +98,27 @@ export class FormDirtiness {
   }
 
   /**
-   * Adds the `amp-form-dirty` class when there are dirty fields and the form
-   * is not being submitted, otherwise removes the class.
+   * Adds or removes the `amp-form-dirty` class and dispatches a
+   * `FORM_DIRTINESS_CHANGE` event that reflects the current dirtiness state,
+   * when the form dirtiness state changes. Does nothing otherwise.
    * @private
    */
-  updateDirtinessClass_() {
+  maybeUpdateClassAndDispatchEvent_() {
```

Comment:
In general we (mainly Su) don't like the the word "maybe" in helper names, because if doesn't convey any information on when that will happen.
UpdateClassAndDispatchEventIfDiryStateChanged?